massa-api/src/proxy.rs

```diff
@@ -0,0 +1,212 @@
+//! Middleware that proxies requests at a specified URI to internal
+//! RPC method calls.
+
+use hyper::header::{ACCEPT, CONTENT_TYPE};
+use hyper::http::HeaderValue;
+use hyper::{Body, Method, Request, Response, StatusCode, Uri};
+use jsonrpsee::core::error::Error as RpcError;
+use std::error::Error;
+use tower::{Layer, Service};
+// use std::error::Error;
+use jsonrpsee::core::params::ArrayParams;
+use jsonrpsee::core::traits::ToRpcParams;
+use jsonrpsee::types::{Id, RequestSer};
+use serde_json::Value;
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use urlencoding::decode;
+
+pub const URI_ADDRESSES: &str = "/addresses";
```

Comment:
We should pay attention to URL encoding for example: 

`%2Faddresses` is the same as `/addresses`